src/elementary/ReactionNumber/src/lookup.hpp

```diff
@@ -0,0 +1,36 @@
+/**
+ *  @brief Private helper function to convert a user provided string into a
+ *         reaction number
+ */
+static Number lookup( const std::string& string ) {
+
+  try {
+
+    return ReactionNumber::conversion_dictionary.at( tolower( string ) );
+  }
+  catch ( ... ) {
```

Comment:
Perhaps we should catch something more specific here (`std::out_of_range`), otherwise the message to `invalid_argument` may be misleading

Reply:
The only possible exception at this point is indeed std::out_of_range but I don't see what would be misleading about the message. This lookup function converts a string to an MT number that is registered in the map. Granted, not all relevant mt numbers are in there yet, but the message is clear enough about it.

Reply:
I'm always worried about things like "the only possible outcome is…" That always gets me into trouble. I agree that the message corresponds nicely with the `std::out_of_range`. I sorry that perhaps `tolower` will throw some other exception. Or sometime in the future we will have additional functionality and forget to change the `catch` block.

Reply:
I'm pretty sure that tolower does not throw, it even works on empty strings but yes, I may overlook some things.